core/src/banking_stage/packet_receiver.rs

```diff
@@ -8,50 +8,81 @@ use {
         unprocessed_transaction_storage::UnprocessedTransactionStorage,
     },
     crossbeam_channel::RecvTimeoutError,
-    solana_measure::measure::Measure,
-    solana_sdk::{
-        saturating_add_assign,
-        timing::{duration_as_ms, timestamp},
-    },
-    std::{
-        sync::atomic::Ordering,
-        time::{Duration, Instant},
-    },
+    solana_measure::{measure, measure_us},
+    solana_sdk::{saturating_add_assign, timing::timestamp},
+    std::{sync::atomic::Ordering, time::Duration},
 };
 
 pub struct PacketReceiver;
 
 impl PacketReceiver {
-    #[allow(clippy::too_many_arguments)]
     /// Receive incoming packets, push into unprocessed buffer with packet indexes
     pub fn receive_and_buffer_packets(
         packet_deserializer: &mut PacketDeserializer,
-        recv_start: &mut Instant,
-        recv_timeout: Duration,
         id: u32,
         unprocessed_transaction_storage: &mut UnprocessedTransactionStorage,
         banking_stage_stats: &mut BankingStageStats,
         tracer_packet_stats: &mut TracerPacketStats,
         slot_metrics_tracker: &mut LeaderSlotMetricsTracker,
     ) -> Result<(), RecvTimeoutError> {
-        let mut recv_time = Measure::start("receive_and_buffer_packets_recv");
-        let ReceivePacketResults {
+        let (result, recv_time_us) = measure_us!({
+            let recv_timeout = Self::get_receive_timeout(unprocessed_transaction_storage);
+            let receive_packet_results = packet_deserializer.handle_received_packets(
+                recv_timeout,
+                unprocessed_transaction_storage.max_receive_size(),
+            )?;
+
+            Self::buffer_packets(
+                receive_packet_results,
+                id,
+                unprocessed_transaction_storage,
+                banking_stage_stats,
+                tracer_packet_stats,
+                slot_metrics_tracker,
+            );
+
+            Ok(())
+        });
+
+        banking_stage_stats
```

Comment:
Functional change here, in that previously this timer was not updated in the case of an `RecvTimeoutError`.

I think it's more correct to always update this, even if it times out.

Reply:
yikes. so current code was double-measuring inside& outside `receive_and_buffer_packets()`? yet another clean up pr justification. lol

putting that aside, i don't think counting `RecvTimeoutError` here makes metrics more useful..

see this

(1) `banking_stage-loop-stats` (NOT counting timeouts):

![image](https://user-images.githubusercontent.com/117807/213899347-3533bd86-ebea-42ab-95d9-5920934d6ea1.png)

(2) `banking_stage-leader_slot_loop_timings` (counting timeouts):

![image](https://user-images.githubusercontent.com/117807/213899356-1804bb0f-9f35-442e-8506-9f83f778dddd.png)

due to metrics inconsistency in current exsisting code, we can see what would be like for the case of counting timeout cases ref: (2)

as this is effectively the sole sleeper  for the banking stage's main loop, it's always saturates to targeted poh time of a single slot _when including timeouts_

i think these metrics were insturumented to diagnose any bad system perf, if any. so, better off not including timeout so that the metrics can be actionable in times like these.

Reply:
hmm, moreover i think we actually should exclude the waited time inside crossbeam_channel's `.recv_timeout()`..? so, `Measure::start()` right after returning from the fn?

Reply:
I think I'll back out these changes and leave the metrics as they were. Can follow-up with further metric changes.

Might want to have finer measurements - I do think it's useful to include the timeout/waiting time **somewhere**, but it's also useful to know how long was spent actually handling those received packets.